File: frontend/src/app/types/dto/person/person-overview.ts
import { PersonData } from "./person-data";
import { PersonAddressDtoId } from "./related/reference/contact/person-address-dto-id";
import { PersonCountryDtoId } from "./related/reference/contact/person-country-dto-id";
import { PersonEmailAddressDtoId } from "./related/reference/contact/person-email-address-dto-id";
import { PersonPhoneNumberDtoId } from "./related/reference/contact/person-phone-number-dto-id";
import { PersonAdditionalHardSkillDtoId } from "./related/reference/skill/person-additional-hard-skill-dto-id";
import { PersonDegreeDtoId } from "./related/reference/skill/person-degree-dto-id";
import { PersonProfessionQualificationDtoId } from "./related/reference/skill/person-profession-qualification-dto-id";
import { PersonSoftSkillDtoId } from "./related/reference/skill/person-soft-skill-dto-id";

export interface PersonOverview {
    person: PersonData;

    addresses: PersonAddressDtoId[];

    phones: PersonPhoneNumberDtoId[];

    emails: PersonEmailAddressDtoId[];

    nationalities: PersonCountryDtoId[];

    degrees: PersonDegreeDtoId[];

    professions: PersonProfessionQualificationDtoId[];

    additionalSkills: PersonAdditionalHardSkillDtoId[];

    softSkills: PersonSoftSkillDtoId[];
}

File: backend/src/main/java/ibee/webapp/todo_app/features/person/dto/PersonOverview.java
package ibee.webapp.todo_app.features.person.dto;


import java.util.List;

import ibee.webapp.todo_app.features.person.related.referenceIds.contact.PersonAddressDtoId;
import ibee.webapp.todo_app.features.person.related.referenceIds.contact.PersonCountryDtoId;
import ibee.webapp.todo_app.features.person.related.referenceIds.contact.PersonEmailAddressDtoId;
import ibee.webapp.todo_app.features.person.related.referenceIds.contact.PersonPhoneNumberDtoId;
import ibee.webapp.todo_app.features.person.related.referenceIds.skill.hard.PersonAdditionalHardSkillDtoId;
import ibee.webapp.todo_app.features.person.related.referenceIds.skill.hard.PersonDegreeDtoId;
import ibee.webapp.todo_app.features.person.related.referenceIds.skill.hard.PersonProfessionQualificationDtoId;
import ibee.webapp.todo_app.features.person.related.referenceIds.skill.soft.PersonSoftSkillDtoId;
/*
* This is the initial response for the Person screen.
* The important point is:
* PersonOverview does not contain all addresses, 
* phones, degrees, etc.
* Those are loaded when their main accordion section 
* is opened.
 */
import ibee.webapp.todo_app.features.person.related.referenceIds.workExp.PersonWorkExperienceDtoId;

public record PersonOverview(

        PersonData person,

        List<PersonAddressDtoId> addresses,

        List<PersonPhoneNumberDtoId> phones,

        List<PersonEmailAddressDtoId> emails,

        List<PersonCountryDtoId> nationalities,

        List<PersonDegreeDtoId> degrees,

        List<PersonProfessionQualificationDtoId> professions,

        List<PersonAdditionalHardSkillDtoId> additionalSkills,

        List<PersonSoftSkillDtoId> softSkills,

        List<PersonWorkExperienceDtoId> workExp

) {
}
